cmd/resourcegen/main.go

```diff
@@ -84,52 +84,53 @@ type config struct {
 	WithHelp bool
 }
 
-func (c config) ResourceLetter() string {
-	return strings.ToLower(string(c.ResourceName[0]))
+func (c config) FullResourceName() string {
+	prefix := c.CloudProvider
+	if c.CloudProvider == "azure" {
+		prefix = "azurerm"
+	}
+
+	return fmt.Sprintf("%s_%s", prefix, c.Filename)
 }
 
-func (c config) CloudProviderTerraformBlock() string {
-	switch c.CloudProvider {
-	case "aws":
-		return `provider "aws" {
-  region                      = "us-east-1"
-  skip_credentials_validation = true
-  skip_metadata_api_check     = true
-  skip_requesting_account_id  = true
-  skip_get_ec2_platforms      = true
-  skip_region_validation      = true
-  access_key                  = "mock_access_key"
-  secret_key                  = "mock_secret_key"
+func (c config) RFuncName() string {
+	return "new" + c.ResourceName
 }
-`
-	case "google":
-		return `provider "google" {
-  credentials = "{\"type\":\"service_account\"}"
-  region      = "us-central1"
+
+func (c config) RegistryFuncName() string {
+	return "get" + c.ResourceName + "RegistryItem"
```

Comment:
Not a big thing here but I realise this will be different for azure. As the azure registry seems to have a `getAzureRM`  convention. I think we should probably remove that convention though.

Reply:
Yep, I thought to use `AzureRM` here too, but AWS/Google don't do that. And as we make the lowercased function names a default now, they will be internal for `azure` package and `AzureRM` part became kinda redundant.